koku/api/provider/view.py

```diff
@@ -142,15 +142,19 @@ def get_queryset(self):
     def create(self, request, *args, **kwargs):
         """Create a Provider."""
         provider_type = request.data.get('type')
-        if provider_type and Provider.PROVIDER_CASE_MAPPING.get(provider_type.lower()):
+        if (provider_type
+                and Provider.PROVIDER_CASE_MAPPING.get(provider_type.lower())
+                and provider_type != provider_type.lower()):
             request.data['type'] = request.data.get('type', '').lower()
```

Comment:
Should this actually be:
```
request.data['type'] = Provider.PROVIDER_CASE_MAPPING.get(provider_type.lower())
```

Reply:
The HTML form sends the `request.data` as a QueryDict which is immutable, so the `request.data['type'] = something` was throwing an AttributeError.

However, I agree that we should probably use the case_mapping here anyway.

Reply:
I take that back. We explicitly set provider_type to lowercase here, and then the serializer `create` gets the DB expected value. So here, it should be lowercase...unless we change the serializer.

Reply:
@maskarb  first of all - thanks for fixing this! I didn't realize my PR had broken the browsable api because I was using postman to test! Secondly ^ you're correct in how I do this. I transformed the type to lowercase in the view if it was populated and in the mapping table because the serializer expects it to be lowercase. I then did the change to the actual DB expected value in the serializer 👍